simulator/infra/log.h

```diff
@@ -49,6 +50,9 @@ class Log
     mutable LogOstream sout;
     mutable LogOstream serr;
 
+    std::ofstream& jsonout();
+    bool jsonout_enabled = false;
```

Comment:
By using `LogOstream` you may omit the boolean variable, and dump data unconditionally.